app/src/main/java/com/smassive/stararchwars/films/view/FilmsActivity.kt

```diff
@@ -51,7 +58,7 @@ class FilmsActivity : AppCompatActivity() {
 
   private fun configViewModel() {
     filmsViewModel = ViewModelProviders.of(this).get(FilmsViewModel::class.java)
-    filmsViewModel.init(FilmsRepository(FilmsFirebaseSource(), asApp().roomDb.filmDao()))
+    filmsViewModel.init(filmsRepository)
```

Comment:
I think you could inject filmsViewModel directly, I always prefer passing dependencies in instantiate time, why you don't pass filmsRepository on the constructor?